File: NET_4.5.2/C1.WPF.FlexGrid/CS/CustomTypeDescriptor/CellFactory/FinancialCellFactory.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Data;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Collections.Generic;
using C1.WPF.FlexGrid;

namespace CustomTypeDescriptor
{
    public class FinancialCellFactory : CellFactory
    {
        static Thickness _thicknessEmpty = new Thickness(0);
        
        // bind cell to ticker
        public override void CreateCellContent(C1FlexGrid grid, Border bdr, CellRange range)
        {
            // create binding for this cell
            var r = grid.Rows[range.Row];
            var c = grid.Columns[range.Column];
            var propName = c.BoundPropertyName;
            if (r.DataItem is FinancialData &&
               (propName == "LastSale" || propName == "Bid" || propName == "Ask"))
            {
                StockTicker ticker = new StockTicker();
                bdr.Child = ticker;
                bdr.Padding = _thicknessEmpty;

                // to show sparklines
                ticker.Tag = r.DataItem;
                ticker.BindingSource = propName;

                // traditional binding
                var binding = new Binding(propName);
                binding.Source = r.DataItem;
                binding.Mode = BindingMode.OneWay;
                ticker.SetBinding(StockTicker.ValueProperty, binding);
            }
            else
            {
                // use default implementation
                base.CreateCellContent(grid, bdr, range);
            }
        }
    }
}

File: NET_4.6.2/C1.WPF.FlexGrid/VB/CustomTypeDescriptor/CellFactory/FinancialCellFactory.vb
Imports System.Net
Imports System.Windows.Media.Animation
Imports C1.WPF.FlexGrid

Namespace CustomTypeDescriptor
    Public Class FinancialCellFactory
        Inherits CellFactory
        Private Shared _thicknessEmpty As New Thickness(0)

        ' bind cell to ticker
        Public Overrides Sub CreateCellContent(ByVal grid As C1FlexGrid, ByVal bdr As Border, ByVal range As CellRange)
            ' create binding for this cell
            Dim r = grid.Rows(range.Row)
            Dim c = grid.Columns(range.Column)
            Dim name = c.BoundPropertyName
            If TypeOf r.DataItem Is FinancialData AndAlso (name = "LastSale" OrElse name = "Bid" OrElse name = "Ask") Then
                Dim ticker As New StockTicker()
                bdr.Child = ticker
                bdr.Padding = _thicknessEmpty
                ' to show sparklines
                ticker.Tag = r.DataItem
                ticker.BindingSource = name

                ' traditional binding
                Dim binding_Renamed = New Binding(name)
                binding_Renamed.Source = r.DataItem
                binding_Renamed.Mode = BindingMode.OneWay
                ticker.SetBinding(StockTicker.ValueProperty, binding_Renamed)
            Else
                ' use default implementation
                MyBase.CreateCellContent(grid, bdr, range)
            End If
        End Sub
    End Class
End Namespace
